Projects/Server/ExpansionInfo.cs

```diff
@@ -188,21 +192,89 @@ public static string GetEraFolder(string parentDirectory)
         return null;
     }
 
+    public static void StoreMapSelection(MapSelection mapSelection, Expansion expansion)
+    {
+        int expansionIndex = (int)expansion;
+        Table[expansionIndex].MapSelectionFlags = mapSelection.Flags;
+        expansionConfigs[expansionIndex].MapSelectionFlags = mapSelection.Flags;
+    }
+
+    public static void SaveConfiguration(Expansion currentExpansion)
+    {
+        if (expansionConfigs == null)
+        {
+            Console.WriteLine("Failed to write expansion.json due to expansion configs not being loaded");
+            return;
+        }
+
+        ExpansionConfig? matched = null;
+
+        // Find the ExpansionConfig for the currently selected expansion and serialise it                
+        for (var i = 0; i < expansionConfigs.Count; i++)
+        {
+            var shortName = ((Expansion)i).ToString();
+            if (shortName == currentExpansion.ToString())
+            {
+                matched = expansionConfigs[i];
+                break;
+            }
+        }
+        if (matched == null)
+        {
+            Console.WriteLine("Failed to write expansion.json due to current expansion being unmatched");
+            return;
+        }
+
+        JsonConfig.Serialize("Configuration/expansion.json", matched);
+    }
+
+    public static void LoadConfiguration(Expansion currentExpansion)
+    {
+        // As we are replacing an existing expansion configuration, we need to make sure
+        // that the default ones have been loaded
+        if (expansionConfigs == null)
+        {
+            Console.WriteLine("Failed to load expansion.json due to expansion configs not being loaded");
+            return;
```

Comment:
maybe I should just throw an exception here instead....

Reply:
If they deleted expansion.json, we should just generate one from modernuo.json/expansions.json.

If they are missing expansions.json, then throw. And all of this logic should be executed _after_ modernuo.json is generated so we don't have an order of operations issue.

Reply:
Ah yep, cool, I'll do that 🙂